### ISSUE TITLE
Ring does not stop

### ISSUE THREAD

**Issue by Lao-Ax** [RocketChat/Rocket.Chat.ReactNative#5424]
### Description:

Ring does not stop even after moving to JITSI screen.
See /app/containers/Ringer/index.tsx - first useEffect starts acync operation. It could be done quite **after** the moment when the component was unmounted. On my android device it happens very often.

Excuse me, that I do not propose a PR. As a simple solution could be the follwoing:
1. introduce new ref: `const mounted = useRef(false);` Set it to `true` in the first hook and set it `false` on unmount.
2. Wrap 'play' functions using the const:
```
if (mounted.current) {
  sound.current = expo.sound;
  await sound.current.playAsync();  
  await sound.current.setIsLoopingAsync(true);
}
```
### Environment Information:

- Rocket.Chat Server Version: 6.4.0
- Rocket.Chat App Version: 4.40+; 4.44.0
- Device Name: Huawei P10
- OS Version: Android 9
- videconf provider: jitsi (type = videoconf)

### Steps to reproduce:

1. Go to any chat
2. Start a call
3. Press "call"

### Expected behavior:

Ring stops when jitsi meet screen is opened

### Actual behavior:

Ring constatnly rings.

**Comment by brf153** [comment]
@Lao-Ax you get a ring when someone makes a call in the room?

**Comment by brf153** [comment]
I don't get any

**Comment by Lao-Ax** [comment]
@brf153 
I do have an instant ring when I make a call to anybody. I didn't test a case for incoming call. Will do later.

**Comment by GleidsonDaniel** [comment]
> @Lao-Ax you get a ring when someone makes a call in the room?

There is only a ringer when the call is direct, in group ou team rooms there is no ringer because it is not part of the planning to call several people at the same time, imagine a room with 100 people, 100 people receiving a call at the same time.

**Comment by brf153** [comment]
@GleidsonDaniel I believe Rocket Chat may have deprecated the feature for creating direct chats when using the open Rocket Chat server. Is there a workaround to initiate direct chats? I attempted to resolve this by cloning the server repository and concurrently running the backend and mobile repositories, but my laptop became unresponsive.

**Comment by Lao-Ax** [comment]
@brf153 
Hi, you asked me:
> get a ring when someone makes a call in the room

No, I don't have any. I've also tested the latest version #44: there is also no ring on an incoming call in direct messages neither in channels.

Looks like it is another issue which is worth to fix someday.

**Comment by GleidsonDaniel** [comment]
@Lao-Ax 
A confirmation, are you talking about the music playing or the notification above within the app about an incoming call?

**Comment by Lao-Ax** [comment]
@GleidsonDaniel 
> are you talking about the music playing

I'm talking about this, you are right. When you start a call, the redux state prop named 'calling' (see /app/lib/hooks/useVideoConf/StartACallActionSheet.tsx#L45) become 'true' in some a saga. Then the Ringer component is rendered, and it starts async task.

**Comment by Lao-Ax** [comment]
Hi! I've just tested the fix of the Ringer conponent. 
I don't like to say this, but the issue wasn't fix. The root cause of the error still the same: the async task of starting a music could be finished a bit after when component was unmounted.

A video below (with audio) is how it looks on my device even with the fix #5453.

https://github.com/RocketChat/Rocket.Chat.ReactNative/assets/33345177/639e9fed-5068-4b85-a151-a01072ca5c55
